manifests/init.pp

```diff
@@ -1,64 +1,64 @@
-  $ipxe_filename                                          = undef,
-  $ipxe_bootstrap                                         = undef,
-  String $logfacility                                     = 'daemon',
-  Integer $default_lease_time                             = 43200,
-  Integer $max_lease_time                                 = 86400,
-  $service_ensure                                         = running,
-  $globaloptions                                          = '',
-  Optional[Stdlib::Port] $omapi_port                      = undef,
-  Optional[String] $omapi_name                            = undef,
-  String $omapi_algorithm                                 = 'HMAC-MD5',
-  Optional[String] $omapi_key                             = undef,
-  Boolean $authoritative                                  = true,
-  Variant[Array[String],String[1]] $extra_config          = [],
-  $dhcp_dir                                               = $dhcp::params::dhcp_dir,
-  String $dhcpd_conf_filename                             = 'dhcpd.conf',
-  $packagename                                            = $dhcp::params::packagename,
-  Boolean $manage_package                                 = true,
-  $servicename                                            = $dhcp::params::servicename,
-  Boolean $manage_service                                 = true,
-  $package_provider                                       = $dhcp::params::package_provider,
-  Integer[0,65535] $ldap_port                             = 389,
-  String $ldap_server                                     = 'localhost',
-  String $ldap_username                                   = 'cn=root, dc=example, dc=com',
-  String $ldap_password                                   = '',
-  String $ldap_base_dn                                    = 'dc=example, dc=com',
-  Enum['dynamic', 'static'] $ldap_method                  = 'dynamic',
-  Optional[Stdlib::Absolutepath] $ldap_debug_file         = undef,
-  Boolean $use_ldap                                       = false,
-  String $option_code150_label                            = 'pxegrub',
-  String $option_code150_value                            = 'text',
-  Hash[String, Hash] $dhcp_classes                        = {},
-  Hash[String, Hash] $hosts                               = {},
-  Hash[String, Hash] $ignoredsubnets                      = {},
-  Hash[String, Hash] $pools                               = {},
-  Hash[String, Hash] $pools6                              = {},
-  Optional[Stdlib::Absolutepath] $dhcpd_binary            = $dhcp::params::dhcpd_binary
+  Optional[Array[String[1]]] $dnsdomain                            = undef,
+  Array[Stdlib::IP::Address::V4] $nameservers                      = [],
+  Array[Stdlib::IP::Address::V6] $nameservers_ipv6                 = [],
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]] $ntpservers     = [],
+  Array[Stdlib::Fqdn] $dnssearchdomains                            = [],
```

Comment:
I think this is too restrictive as it stops putting a top level domain in.

Reply:
will update it to `String[1]` than.